### PR TITLE
Added EggDroppingPuzzle.java

### PR DESCRIPTION
**Pull request by Nidakhan786**
**Fixes issue:**
N/A

**Changes:**
Added EggDroppingPuzzle.java

### CODE REVIEW

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+//Dynamic Programming solution for the Egg Dropping Puzzle
+ /* Returns the minimum number of attempts
+  needed in the worst case for a eggs and b floors.*/
+
+import java.util.*;
+public class EggDroppingPuzzle
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Add newline above, remove newline below.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		for (int i = 1; i <= a; ++i)
+		{
+		    eggFloor[i][0] = 0;   // Zero trial for zero floor.
+		    eggFloor[i][1] = 1;   // One trial for one floor 
+		}
+		
+	        // j trials for only 1 egg
+		
+                for (int j = 1; j <= b; ++j)
+			 eggFloor[1][j] = j;
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Fix indentation, which should be just one tab.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		  {
+			for (int j = 2; j <= b; ++j)
+			{
+				eggFloor[i][j] = Integer.MAX_VALUE;
+				for (x = 1; x <= j; ++x)
+				{   // get worst case from:case in which egg break (eggFloor[i-1][x-1]) and  case in which egg does not break (eggFloor[i][j-x]).
+					result = 1 + Math.max(eggFloor[i - 1][x - 1], eggFloor[i][j - x]);  
+          
+					//choose min of all values for particular x
+        				 if (result < eggFloor[i][j])
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Fix indentation inconsistencies.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			}
+		}
+		  
+		return eggFloor[a][b];
+	}
+		
+	//testing the program
+	public static void main(String args[])
+	{   
+	Scanner	sc = new Scanner(System.in);
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Indentation.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	System.out.println("Enter no. of eggs");
+	
+		int a = Integer.parseInt(sc.nextLine());
+		System.out.println("Enter no. of floors");
+		
+		int b = Integer.parseInt(sc.nextLine());
+              //result outputs min no. of trials in worst case for a eggs and b floors
+                int result = minTrials(a, b);
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Indentation.

Comment (Nidakhan786) [reply code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: okk 👍

**Comment by Nidakhan786** [comment]
@arnavb please review the changes .. 😄

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
+
+//Dynamic Programming solution for the Egg Dropping Puzzle
+ /* Returns the minimum number of attempts
+  needed in the worst case for a eggs and b floors.*/
+import java.util.*;
+public class EggDroppingPuzzle
+ 	{
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Er, Indentation. Braces should be on the same level as their descriptors.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java (hunk 5 above)
Comment (arnavb) [reply code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Remove the extra space before the int.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java (hunk 6 above)
Comment (Nidakhan786) [reply code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Done

**Comment by Nidakhan786** [comment]
@arnavb is it fine now?

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
+
+//Dynamic Programming solution for the Egg Dropping Puzzle
+ /* Returns the minimum number of attempts
+  needed in the worst case for a eggs and b floors.*/
+import java.util.*;
+public class EggDroppingPuzzle
+ {
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Remove extra space here.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
+
+//Dynamic Programming solution for the Egg Dropping Puzzle
+ /* Returns the minimum number of attempts
+  needed in the worst case for a eggs and b floors.*/
+import java.util.*;
+public class EggDroppingPuzzle
+ {    
+    	// min trials with a eggs and b floors 
+    	 private static int minTrials(int a, int b)
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Remove extra space.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+public class EggDroppingPuzzle
+ {    
+    	// min trials with a eggs and b floors 
+    	 private static int minTrials(int a, int b)
+	{
+		int eggFloor[][] = new int[a + 1][b + 1];
+		int result, x;
+		
+		for (int i = 1; i <= a; ++i)
+			{
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Move braces to the for loop.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+		    	eggFloor[i][1] = 1;   // One trial for one floor 
+			}
+		
+	       	 // j trials for only 1 egg
+		
+                 for (int j = 1; j <= b; ++j)
+	         	eggFloor[1][j] = j;
+		
+	
+   		 for (int i = 2; i <= a; ++i)
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Fix indentation

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+					 	eggFloor[i][j] = result;
+				}
+			}
+		  }
+		  
+		return eggFloor[a][b];
+	}
+		
+	//testing the program
+	public static void main(String args[])
```
Comment (arnavb) [inline code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: Fix indentation of braces.

File: code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java (hunk 7 above)
Comment (Nidakhan786) [reply code/dynamic_programming/src/egg_dropping_puzzle/EggDroppingPuzzle.java]: okk done

**Comment by arnavb** [comment]
@Nidakhan786 There are still _a lot of indentation issues_. Look over your file, line by line, and see what they are!

You know what, let me just do it for you:

```Java
//Dynamic Programming solution for the Egg Dropping Puzzle
/* Returns the minimum number of attempts
 needed in the worst case for a eggs and b floors.*/
import java.util.*;

public class EggDroppingPuzzle {
	// min trials with a eggs and b floors 
	private static int minTrials(int a, int b) {
		int eggFloor[][] = new int[a + 1][b + 1];
		int result, x;

		for (int i = 1; i <= a; ++i) {
			eggFloor[i][0] = 0; // Zero trial for zero floor.
			eggFloor[i][1] = 1; // One trial for one floor 
		}
		// j trials for only 1 egg	
		for (int j = 1; j <= b; ++j) {
			eggFloor[1][j] = j;
		}
		for (int i = 2; i <= a; ++i) {
			for (int j = 2; j <= b; ++j) {
				eggFloor[i][j] = Integer.MAX_VALUE;
				for (x = 1; x <= j; ++x) {
					// get worst case from:case in which egg break (eggFloor[i-1][x-1]) and  case in which egg does not break (eggFloor[i][j-x]).
					result = 1 + Math.max(eggFloor[i - 1][x - 1], eggFloor[i][j - x]);
					//choose min of all values for particular x
					if (result < eggFloor[i][j])
						eggFloor[i][j] = result;
				}
			}
		}
		return eggFloor[a][b];
	}

	//testing the program
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter no. of eggs");

		int a = Integer.parseInt(sc.nextLine());
		System.out.println("Enter no. of floors");

		int b = Integer.parseInt(sc.nextLine());
		//result outputs min no. of trials in worst case for a eggs and b floors
		int result = minTrials(a, b);
		System.out.println("Minimum number of attempts needed in Worst case with a eggs and b floor are: " + result);
	}
}
```

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Comment by Nidakhan786** [comment]
@AdiChat  Thank You 😄

**Pull request merged**